File: backend/src/main/java/com/tabletennis/app/models/Result.java
package com.tabletennis.app.models;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

@Entity
@Table(name = "results")
public class Result {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    private Long player1Id;
    @NotNull
    private Long player2Id;
    @NotNull
    private int pointsPlayer1;
    @NotNull
    private int pointsPlayer2;
    @NotNull
    private int roundId;
    @NotNull
    private int uniqueGameId;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getPlayer1Id() {
        return player1Id;
    }

    public void setPlayer1Id(Long player1Id) {
        this.player1Id = player1Id;
    }

    public Long getPlayer2Id() {
        return player2Id;
    }

    public void setPlayer2Id(Long player2Id) {
        this.player2Id = player2Id;
    }

    public int getPointsPlayer1() {
        return pointsPlayer1;
    }

    public void setPointsPlayer1(int pointsPlayer1) {
        this.pointsPlayer1 = pointsPlayer1;
    }

    public int getPointsPlayer2() {
        return pointsPlayer2;
    }

    public void setPointsPlayer2(int pointsPlayer2) {
        this.pointsPlayer2 = pointsPlayer2;
    }

    public int getRoundId() {
        return roundId;
    }

    public void setRoundId(int roundId) {
        this.roundId = roundId;
    }

    public int getUniqueGameId() {
        return uniqueGameId;
    }

    public void setUniqueGameId(int uniqueGameId) {
        this.uniqueGameId = uniqueGameId;
    }
}

File: mobile_application/lib/entities/result.dart
class Result {
  int _id;
  int _player1Id;
  int _player2Id;
  int _pointsPlayer1;
  int _pointsPlayer2;
  int _roundId;
  int _uniqueGameId;

  Result(
    this._id,
    this._player1Id,
    this._player2Id,
    this._pointsPlayer1,
    this._pointsPlayer2,
    this._roundId,
    this._uniqueGameId,
  );

  factory Result.fromJson(Map<String, dynamic> json) {
    return Result(
      json['id'] ?? 0,
      json['player1Id'] ?? 0,
      json['player2Id'] ?? 0,
      json['pointsPlayer1'] ?? 0,
      json['pointsPlayer2'] ?? 0,
      json['roundId'] ?? 0,
      json['uniqueGameId'] ?? 0,
    );
  }

  int get id => _id;
  set id(int value) => _id = value;

  int get player1Id => _player1Id;
  set player1Id(int value) => _player1Id = value;

  int get player2Id => _player2Id;
  set player2Id(int value) => _player2Id = value;

  int get pointsPlayer1 => _pointsPlayer1;
  set pointsPlayer1(int value) => _pointsPlayer1 = value;

  int get pointsPlayer2 => _pointsPlayer2;
  set pointsPlayer2(int value) => _pointsPlayer2 = value;

  int get roundId => _roundId;
  set roundId(int value) => _roundId = value;

  int get uniqueGameId => _uniqueGameId;
  set uniqueGameId(int value) => _uniqueGameId = value;
}
